packages/agent/src/agent/http/index.ts

```diff
@@ -747,51 +732,35 @@ export class HttpAgent implements Agent {
       throw TransportError.fromCode(new HttpFetchErrorCode(error));
     }
 
-    const timestamp = response.signatures?.[0]?.timestamp;
-
-    // Skip watermark verification if the user has set verifyQuerySignatures to false
+    // Skip timestamp verification if the user has set verifyQuerySignatures to false
     if (!this.#verifyQuerySignatures) {
       return response;
     }
 
-    if (!timestamp) {
+    const timestampInNs = response.signatures?.[0]?.timestamp;
+    if (!timestampInNs) {
       throw ProtocolError.fromCode(
         new MalformedSignatureErrorCode(
           'Timestamp not found in query response. This suggests a malformed or malicious response.',
         ),
       );
     }
 
-    // Convert the timestamp to milliseconds
-    const timeStampInMs = Number(BigInt(timestamp) / BigInt(1_000_000));
+    const timestampInMs = Number(BigInt(timestampInNs) / BigInt(NANOSECONDS_TO_MSECS));
 
-    this.log.print('watermark and timestamp', {
-      waterMark: this.waterMark,
-      timestamp: timeStampInMs,
-    });
-
-    // If the timestamp is less than the watermark, retry the request up to the retry limit
-    if (Number(this.waterMark) > timeStampInMs) {
-      const error = ProtocolError.fromCode(
-        new TimeoutWaitingForResponseErrorCode(
-          'Timestamp is below the watermark. Retrying query.',
-          requestId,
-        ),
-      );
-      this.log.error('Timestamp is below', error, {
-        timestamp,
-        waterMark: this.waterMark,
+    // If the timestamp is older than the max ingress expiry, retry the request up to the retry limit.
+    // We don't have to consider the #timeDiffMsecs here because the response comes from a single node,
+    // and hence it can be different from the network time.
+    if (Date.now() - timestampInMs > this.#maxIngressExpiryInMs) {
+      this.log.warn('Timestamp is older than the max ingress expiry. Retrying query.', {
+        requestId,
+        timestamp: timestampInMs,
```

Comment:
Could we move this to the subsequent if block? (We don't really retry if we already reached the maximum number of retries?